test/test_nn.py

```diff
@@ -12880,6 +12880,52 @@ def helper(n, c, h, w, ks):
         helper(2, 8, 4, 4, ks=2)
         helper(None, 3, 50, 50, ks=5)
 
+    def test_interpolate_channels_last(self, device):
```

Comment:
there's already test_upsamplingNearest2d (probably also test for bilinear, I didn't check) that checks memory format handling, please don't duplicate the tests and instead refactor so that all the necessary functionality is tested. 
Also, would make sense to test discontiguous (not-memory-dense, for example) inputs/outputs.

Reply:
Seems like `test_upsamplingNearest2d` only checks the output channels-last memory_format but not the backward data.grad memory_format. Do we want to check that as well?

Reply:
Yes, definitely! But it's better when those tests are in one place.